### PR TITLE
(PC-22201) feat(navigation): delete usage of hook navigateToNextScreen

### PR DESCRIPTION
**Pull request by S4andro**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-22201

## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.

### CODE REVIEW

File: src/features/identityCheck/pages/helpers/getCurrentStep.ts [hunk 1]
```diff
@@ -4,4 +4,4 @@ import { SubscriptionRootStackParamList } from 'features/navigation/RootNavigato
 export const getCurrentStep = (
   steps: StepDetails[],
   currentRoute: keyof SubscriptionRootStackParamList
-) => steps.find((step) => step.screens.includes(currentRoute)) ?? null
+) => steps.find((step) => step.firstScreen.includes(currentRoute)) ?? null
```
Comment (voisinhugo) [inline src/features/identityCheck/pages/helpers/getCurrentStep.ts:7]: Ça marche, mais comme `step.firstScreen` n'est plus un array, il vaut mieux vérifier l'égalité non ?
`step.firstScreen === currentRoute`

File: src/features/identityCheck/pages/helpers/useStepperInfo.tsx [hunk 2]
```diff
@@ -23,8 +23,8 @@ export const useStepperInfo = (): {
   subtitle?: string | null
   errorMessage?: string | null
 } => {
-  const { profile } = useSubscriptionContext()
-  const hasSchoolTypes = profile.hasSchoolTypes
+  // const { profile } = useSubscriptionContext()
+  // const hasSchoolTypes = profile.hasSchoolTypes
```
Comment (voisinhugo) [inline src/features/identityCheck/pages/helpers/useStepperInfo.tsx:27]: Attention aux commentaires 😬

File: src/features/identityCheck/pages/helpers/useStepperInfo.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { useGetStepperInfo } from 'features/identityCheck/api/useGetStepperInfo'
 import { usePhoneValidationRemainingAttempts } from 'features/identityCheck/api/usePhoneValidationRemainingAttempts'
 import { IconRetryStep } from 'features/identityCheck/components/IconRetryStep'
 import { IconStepDone } from 'features/identityCheck/components/IconStepDone'
-import { useSubscriptionContext } from 'features/identityCheck/context/SubscriptionContextProvider'
+// import { useSubscriptionContext } from 'features/identityCheck/context/SubscriptionContextProvider'
```
Comment (voisinhugo) [inline src/features/identityCheck/pages/helpers/useStepperInfo.tsx:7]: Attention aux commentaires 😬

File: src/features/identityCheck/pages/helpers/useStepperInfo.native.test.tsx [hunk 4]
```diff
@@ -145,15 +121,17 @@ describe('useStepperInfo', () => {
     const phoneValidationStep = stepsDetails.find(
       (step) => step.name === IdentityCheckStep.PHONE_VALIDATION
     )
-    expect(phoneValidationStep?.screens.includes('PhoneValidationTooManySMSSent')).toEqual(false)
+    expect(phoneValidationStep?.firstScreen.includes('PhoneValidationTooManySMSSent')).toEqual(
+      false
```
Comment (tanguyprache) [inline src/features/identityCheck/pages/helpers/useStepperInfo.native.test.tsx:125]: Pareil ici tester la navigation vers `SetPhoneNumber` et renommer le test

**Review by voisinhugo — APPROVED** [review]

**Pull request merged**